src/BrainPass/BrainPass.sol

```diff
@@ -0,0 +1,402 @@
+    function configureMintLimit(
+        uint256 lowerLimit,
+        uint256 upperLimit
+    ) external onlyOwner {
+        MINT_UPPER_LIMIT = upperLimit;
+        MINT_LOWER_LIMIT = lowerLimit;
+    }
+
+    /// @notice Add a new Pass Type
+    /// @param pricePerDay the price per day of the new pass type
+    /// @param name the name of the new pass type to be added
+    /// @param maxTokens the total number of tokens in the pass
+    /// @param discount the amount in % to be deducted when buying the pass
+    function addPassType(
+        uint256 pricePerDay,
+        string memory name,
+        uint256 maxTokens,
+        uint256 discount
+    ) external onlyOwner whenNotPaused {
+        if (maxTokens <= 0) revert InvalidMaxTokensForAPass();
+        uint256 passId = passIdTracker.current();
+        passTypes[passId] = PassType(
+            passId,
+            name,
+            pricePerDay,
+            maxTokens,
+            discount,
+            0,
+            false
+        );
+        passIdTracker.increment();
+        emit NewPassAdded(passId, name, maxTokens, pricePerDay);
+    }
+
+    /// @notice Toggles the staus of a pass type (paused or unpaused)
+    /// @param passId the Id of the pass
+    function togglePassTypeStatus(
+        uint256 passId
+    ) external onlyOwner whenNotPaused {
+        if (passId >= passIdTracker.current()) revert PassTypeNotFound();
+        PassType storage passType = passTypes[passId];
+        bool newStatus = !passType.isPaused;
+
+        passTypes[passId] = PassType(
+            passType.passId,
+            passType.name,
+            passType.pricePerDay,
+            passType.maxTokens,
+            passType.discount,
+            passType.lastMintedId,
+            newStatus
+        );
+
+        emit PassTypeStatusToggled(passId, passType.name);
+    }
+
+    /// @notice Mint and NFT of a particular passtype
+    /// @param passId The id of the passtype to mint
+    /// @param startTimestamp The time when the NFT subcription time starts
+    /// @param endTimestamp The time when the NFT subcription time ends
+    function mintNFT(
+        uint256 passId,
+        uint256 startTimestamp,
+        uint256 endTimestamp
+    ) external whenNotPaused {
+        if (passId >= passIdTracker.current()) revert PassTypeNotFound();
+        if (addressToNFTPass[msg.sender].tokenId != 0)
+            revert AlreadyMintedAPass();
+        PassType storage passType = passTypes[passId];
+
+        if (passType.isPaused) revert PassTypeIsPaused();
+        if (passType.lastMintedId >= passType.maxTokens)
+            revert PassMaxSupplyReached();
+        if (!validatePassDuration(startTimestamp, endTimestamp))
+            revert DurationNotInTimeFrame();
+
+        uint256 price = calculatePrice(passId, startTimestamp, endTimestamp);
+
+        bool success = IERC20(iqToken).transferFrom(msg.sender, owner(), price);
```

Comment:
lets transfer instead of owner to this contract. and we will withdraw the IQ tokens whenever we want